embedded-tests/src/test/java/org/apache/druid/testing/embedded/query/ITSqlQueryTest.java

```diff
@@ -0,0 +1,379 @@
+        "application/json; charset=UTF-8",
+        "{\"query\":\"select 567\"}",
+        (request) -> {
+        },
+        (statusCode, responseBody) -> {
+          Assertions.assertEquals(200, statusCode);
+          Assertions.assertEquals("[{\"EXPR$0\":567}]", responseBody);
+        }
+    );
+  }
+
+  @ParameterizedTest
+  @FieldSource("flags")
+  public void testInvalidJSONFormat(boolean shouldQueryBroker)
+  {
+    executeQuery(
+        shouldQueryBroker ? brokerEndpoint : routerEndpoint,
+        MediaType.APPLICATION_JSON,
+        "{\"query\":select 567}",
+        (request) -> {
+        },
+        (statusCode, responseBody) -> {
+          Assertions.assertEquals(400, statusCode);
+          assertStringCompare("Malformed SQL query", responseBody, responseBody::contains);
+        }
+    );
+  }
+
+  @ParameterizedTest
+  @FieldSource("flags")
+  public void testEmptyQuery_TextPlain(boolean shouldQueryBroker)
+  {
+    executeQuery(
+        shouldQueryBroker ? brokerEndpoint : routerEndpoint,
+        MediaType.TEXT_PLAIN,
+        null,
+        (request) -> {
+        },
+        (statusCode, responseBody) -> {
+          Assertions.assertEquals(400, statusCode);
+          assertStringCompare("Empty query", responseBody, responseBody::contains);
+        }
+    );
+  }
+
+  @ParameterizedTest
+  @FieldSource("flags")
+  public void testEmptyQuery_UrlEncoded(boolean shouldQueryBroker)
+  {
+    executeQuery(
+        shouldQueryBroker ? brokerEndpoint : routerEndpoint,
+        MediaType.APPLICATION_FORM_URLENCODED,
+        null,
+        (request) -> {
+        },
+        (statusCode, responseBody) -> {
+          Assertions.assertEquals(400, statusCode);
+          assertStringCompare("Empty query", responseBody, responseBody::contains);
+        }
+    );
+  }
+
+  @ParameterizedTest
+  @FieldSource("flags")
+  public void testBlankQuery_TextPlain(boolean shouldQueryBroker)
+  {
+    executeQuery(
+        shouldQueryBroker ? brokerEndpoint : routerEndpoint,
+        MediaType.TEXT_PLAIN,
+        "     ",
+        (request) -> {
+        },
+        (statusCode, responseBody) -> {
+          Assertions.assertEquals(400, statusCode);
+          assertStringCompare("Empty query", responseBody, responseBody::contains);
+        }
+    );
+  }
+
+  @ParameterizedTest
+  @FieldSource("flags")
+  public void testEmptyQuery_JSON(boolean shouldQueryBroker)
+  {
+    executeQuery(
+        shouldQueryBroker ? brokerEndpoint : routerEndpoint,
+        MediaType.APPLICATION_JSON,
+        null,
+        (request) -> {
+        },
+        (statusCode, responseBody) -> {
+          Assertions.assertEquals(400, statusCode);
+          assertStringCompare("Empty query", responseBody, responseBody::contains);
+        }
+    );
+  }
+
+  /**
+   * When multiple Content-Type headers are set, the first one (in this case, it's the text format) should be used.
+   */
+  @ParameterizedTest
+  @FieldSource("flags")
```

Comment:
Rather than have all the tests be parameterized, I wonder if it would be cleaner to:
- write the main test class from point-of-view of Broker.
- then add a separate test class (using junit5 `@Nested` or even a separate class) which does testing form PoV of Router.